### PR TITLE
Attack vectors now work, fixes numerous bugs with unarmed weapons

### LINKED ISSUE #56138: Worn unarmed weapons no longer add damage to attacks
### Describe the bug

Weapons that are *worn* to increase your punch damage, such as skewer knuckles, do not add their melee damage to your strike. If you give them UNARMED_WEAPON and then *wield* them, then it works.

### Steps to reproduce

1. Add UNARMED_WEAPON to skewer knuckles.
2. Spawn zombie.
3. Punch zombie with knuckles worn as intended, see that your strikes are same as empty-handed(no bleeding or increased damage).
4. Wield skewer knuckles & slap zombie with them, see bleeding + increased damage.

### Expected behavior

Not needing to slap enemies with my knuckles since they were intended to worn to punch since #54313.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22000.556 (21H2)
- Game Version: e033309 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

This was tested with both Brawling and multiple martial arts to make sure I wasn't just using a technique with an inapplicable attack vector.

### PR DESCRIPTION
**Pull request by MylieDaniels**
#### Summary
Infrastructure "Attack vectors (headbutts, kicks, etc) work, bugfixes for unarmed weapons"

#### Purpose of change
Fixes #56138
There were a number of issues with unarmed weapons (typically this meant hitting something with your armor).

#### Describe the solution
Refactor attack_vectors into an enum, fix inverted logic with unarmed vs weapon skills, unify logic for attack_vectors.
Large changes such that a melee attack now uses a weighted list to decide which vector to attack with, prioritizing more capable body parts, with a threshold that prevents terrible attacks from being made except when they have to be.
Splits leg and arm character_modifiers for melee, allowing for broken arms to only moderately impair kicks, etc.

#### Testing
Tested half the martial arts, the ones that have the most varied attack_vectors.  I was able to kick a zombie to death with broken arms using Muay Thai, which was generally speaking impossible before.

### CODE REVIEW

**Comment by Maleclypse** [comment]
@MylieDaniels can you check your github settings and see if github has flagged you.  It doesn't look like any tests are running against your PRs and people before and after have test results.

**Comment by MylieDaniels** [comment]
> @MylieDaniels can you check your github settings and see if github has flagged you. It doesn't look like any tests are running against your PRs and people before and after have test results.

I don't appear to be flagged in any way, I'm unsure what's happening.

File: src/character.cpp [hunk 1]
```diff
@@ -497,6 +497,36 @@ std::string enum_to_string<blood_type>( blood_type data )
     cata_fatal( "Invalid blood_type" );
 }
 
+template<>
+std::string enum_to_string<attack_vector>( attack_vector data )
+{
+    switch( data ) {
+            // *INDENT-OFF*
```
Comment (Maleclypse) [inline src/character.cpp:504]:
```suggestion
            // *INDENT-OFF*
            // Test kick test delete
```

**Comment by Maleclypse** [comment]
Ok me adding the extra comment line seems to have kicked the tests off.  More github weirdness.  Hopefully it will run tests for you now after you delete my comment in your code.

**Comment by Termineitor244** [comment]
I'm not sure if related, but does this fix #56138?

**Comment by MylieDaniels** [comment]
Yes, this fixes #56138.

**Comment by MylieDaniels** [comment]
Converting to draft so I can spend a day testing to ensure damage values are correct.

**Comment by MylieDaniels** [comment]
If this error occurred on my local machine, ever, I would be less frustrated 💢

**Comment by TheShadowFerret** [comment]
This will be obsoleted by #57972

**Pull request closed without merging**